src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -913,6 +965,7 @@ public void Close()
         /// <internalonly/>
         private void EnsureState(State state)
         {
+            CheckDisposed();
```

Comment:
Since you've already added the disposal check here, most of the other checks are redundant.